pkg/template/event_test.go

```diff
@@ -315,6 +316,61 @@ func Test_NewResources(t *testing.T) {
 	}
 }
 
+func TestResolveParams(t *testing.T) {
+	tests := []struct {
+		name     string
+		body     []byte
+		params   []pipelinev1.ParamSpec
+		bindings []*triggersv1.TriggerBinding
+		want     []pipelinev1.Param
+	}{{
+		name: "values with newlines",
+		body: json.RawMessage(`{"foo": "bar\r\nbaz"}`),
+		params: []pipelinev1.ParamSpec{{
+			Name: "param1",
+		}, {
+			Name: "param2",
+		}},
+		bindings: []*triggersv1.TriggerBinding{
+			bldr.TriggerBinding("tb", "namespace",
+				bldr.TriggerBindingSpec(
+					bldr.TriggerBindingParam("param1", "qux"),
+					bldr.TriggerBindingParam("param2", "$(body.foo)"),
+				),
+			),
+		},
+		want: []pipelinev1.Param{{
+			Name: "param1",
+			Value: pipelinev1.ArrayOrString{
+				StringVal: "qux",
+				Type:      pipelinev1.ParamTypeString,
+			},
+		}, {
+			Name: "param2",
+			Value: pipelinev1.ArrayOrString{
+				StringVal: "bar\\r\\nbaz",
+				Type:      pipelinev1.ParamTypeString,
+			},
+		}},
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			params, err := ResolveParams(tt.bindings, tt.body, map[string][]string{}, tt.params)
+			if err != nil {
+				t.Fatalf("ResolveParams() returned unexpected error: %s", err)
+			}
+			trans := cmp.Transformer("Sort", func(in []pipelinev1.Param) []pipelinev1.Param {
```

Comment:
Could we use `cmpopts.SortSlices` here as we do in `resource_test.go`?
https://github.com/tektoncd/triggers/blob/5d31c9600537b0dc5b89f77eded8c1ecdc861c52/pkg/template/resource_test.go#L117

Reply:
ah kk that's simpler, i didnt know about that!! only downside i can think of is that'll sort any slice, not just pipeline params? maybe thats actually a good thing - usually we don't care about order

Reply:
> only downside i can think of is that'll sort any slice, not just pipeline params? maybe thats actually a good thing - usually we don't care about order

nvm, just saw the value of compareParam! no downsides at all :D